Jenkinsfile

```diff
@@ -37,9 +37,15 @@ node {
         return
     }
 
+
+    echo "Current branch is \"${scm.branches[0].name}\", primary: ${env.BRANCH_IS_PRIMARY}"
     def buildType = 'install'
-    if ("${env.BRANCH_NAME}".equals('develop')) {
+    if (env.BRANCH_IS_PRIMARY && 'develop'.equals(scm.branches[0].name)) {
```

Comment:
@nicolatimeus if you want to detect whether you're on the main branch or in a PR the best way I found is looking at the `CHANGE_TARGET` env variable.

See: https://github.com/eurotech/add-ons-shared-libraries/blob/e00d121a651ab8138b5bfc55577a4153975a78e2/vars/continuousIntegrationPipeline.groovy#L4 and https://jenkins.thetaphi.de/env-vars.html/